src/mlpack/methods/ann/layer/advantage_loss_impl.hpp

```diff
@@ -0,0 +1,75 @@
+/**
+ * @file Advantage_error_impl.hpp
+ * @author Rohan Raj
+ *
+ * Implementation of the Advantage error performance function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ADVANTAGE_ERROR_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_ADVANTAGE_ERROR_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "advantage_loss.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+template<typename InputDataType, typename OutputDataType>
+AdvantageError<InputDataType, OutputDataType>::AdvantageError()
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename InputType, typename TargetType>
+double AdvantageError<InputDataType, OutputDataType>::Forward(
+    const InputType&& input, const TargetType&& target)
+{
+  /*
+  The loss function is mean of log(probability) * target
+  The target captures the advantage of taking the action.
+  The -1 is multiplied to take negative of mean.
+
+  Reference : https://karpathy.github.io/2016/05/31/rl/
```

Comment:
We should remove the armadillo related documentation here, a user should be able to lookup specific operations.